File: server/src/main/java/com/noteblock/server/model/Folder.java
package com.noteblock.server.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "folders")
@Getter
@Setter
@NoArgsConstructor
@RequiredArgsConstructor
public class Folder {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    @NonNull
    private String name;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt = LocalDateTime.now();

    public String getName() {
        return name;
    }
}

File: server/model/folder.go
package model

import (
	"github.com/google/uuid"
	"gorm.io/gorm"
	"time"
)

type Folder struct {
	ID        string `gorm:"type:uuid;primaryKey"`
	Name      string
	ParentID  *string
	CreatedAt time.Time
	UpdatedAt time.Time
}

func (f *Folder) BeforeCreate(tx *gorm.DB) (err error) {
	f.ID = uuid.New().String()
	f.CreatedAt = time.Now()
	f.UpdatedAt = time.Now()
	return
}
